src/Transport/Creation/AzureServiceBusQueueCreator.cs

```diff
@@ -53,53 +53,55 @@ public AzureServiceBusQueueCreator(ReadOnlySettings settings)
                 if (!await ExistsAsync(namespaceManager, description.Path).ConfigureAwait(false))
                 {
                     await namespaceManager.CreateQueue(description).ConfigureAwait(false);
-                    logger.InfoFormat("Queue '{0}' created", description.Path);
+                    logger.InfoFormat("Queue '{0}' created in namespace {1}", description.Path, namespaceManager.Address.Host);
 
-                    await rememberExistence.AddOrUpdate(description.Path, s => Task.FromResult(true), (s, b) => Task.FromResult(true)).ConfigureAwait(false);
+                    var key = GenerateQueueKey(namespaceManager, queuePath);
+
+                    await rememberExistence.AddOrUpdate(key, s => Task.FromResult(true), (s, b) => Task.FromResult(true)).ConfigureAwait(false);
```

Comment:
`Task.FromResult(true)` could be cached statically. You could add for example a TrueTask to the TaskExtensions